src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1169,6 +1175,59 @@ public override bool HandleCall (MethodBody callingMethodBody, MethodReference c
 					}
 					break;
 
+				//
+				// Type.BaseType
+				//
+				case IntrinsicId.Type_get_BaseType: {
+						foreach (var value in methodParams[0].UniqueValues ()) {
+							if (value is LeafValueWithDynamicallyAccessedMemberNode dynamicallyAccessedMemberNode) {
+								DynamicallyAccessedMemberTypes propagatedMemberTypes = DynamicallyAccessedMemberTypes.None;
+								if (dynamicallyAccessedMemberNode.DynamicallyAccessedMemberTypes == DynamicallyAccessedMemberTypes.All)
+									propagatedMemberTypes = DynamicallyAccessedMemberTypes.All;
+								else {
+									if (dynamicallyAccessedMemberNode.DynamicallyAccessedMemberTypes.HasFlag (DynamicallyAccessedMemberTypes.PublicConstructors))
```

Comment:
I don't think we propagate constructors to base types.

Reply:
Good point - and the ones below as well - fixed